src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -205,6 +211,100 @@ get_aie_metadata_info(const xrt_core::device* device)
   return aie_meta;
 }
 
+static void 
+get_bd_core(zynqaie::Aie* aieArray, boost::property_tree::ptree& aie_meta , boost::property_tree::ptree& ptarray, uint8_t& row_offset)
+{
+   boost::property_tree::ptree empty_pt;
+   std::vector<std::pair<int,int>>v;
+   for (auto& gr: aie_meta.get_child("aie_metadata.graphs", empty_pt)) {
+     boost::property_tree::ptree igraph;
+     auto row_it = gr.second.get_child("core_rows").begin();
+     for (const auto& node : gr.second.get_child("core_columns", empty_pt)) {
+       igraph.put("row", row_it->second.data());
+       igraph.put("column", node.second.data());
+       uint8_t r = igraph.get<uint8_t>("row");
+       uint8_t col = igraph.get<uint8_t>("column");
+       uint8_t row = r + row_offset;
+       v.push_back(std::make_pair(r,col));
+       boost::property_tree::ptree bdtree = aieArray->get_bd_info(row, col);
+       std::string st = std::to_string(col) + "_" + std::to_string(r);
+       ptarray.add_child((st + ".buffer-descriptors"),bdtree);
+       row_it++;
+     }    
+  }
+  for (const auto& gr : aie_meta.get_child("aie_metadata.EventGraphs", empty_pt)) {
+     boost::property_tree::ptree igraph;
+     auto dma_row_it = gr.second.get_child("dma_rows", empty_pt).begin();
+     for (const auto& node : gr.second.get_child("dma_columns", empty_pt)) {
+       igraph.put("column", node.second.data());
+       igraph.put("row", dma_row_it->second.data());
+       // Checking whether this core is already added
+       for(const auto& itr: v)
+         if ((itr.first == igraph.get<uint8_t>("row")) && (itr.second == igraph.get<uint8_t>("column")))
+            continue;
+       uint8_t r = igraph.get<uint8_t>("row");
+       uint8_t col = igraph.get<uint8_t>("column");
+       uint8_t row = r + row_offset;
+       boost::property_tree::ptree bdtree = aieArray->get_bd_info(row, col);
+       std::string st = std::to_string(col) + "_" + std::to_string(r);
+       ptarray.add_child((st + ".buffer-descriptors"),bdtree);
+       dma_row_it++;
+     }
+  }
+}
+
+static void 
+get_bd_mem(zynqaie::Aie* aieArray, boost::property_tree::ptree& aie_meta , boost::property_tree::ptree& ptarray, uint8_t& row_offset)
```

Comment:
A get function usually returns some thing, in this case the `ptarray`, but if it is adding to `ptarray` then the function name should reflect this.  Also if `aie_meta` is not modified it should be `const`